streams/src/test/java/org/apache/kafka/streams/integration/PurgeRepartitionTopicIntegrationTest.java

```diff
@@ -0,0 +1,223 @@
+                        && config.get(TopicConfig.SEGMENT_BYTES_CONFIG).value().equals(purgeSegmentBytes.toString());
+            } catch (Exception e) {
+                return false;
+            }
+        }
+    }
+
+    private interface TopicSizeVerifier {
+
+        boolean verify(long currentSize);
+    }
+
+    private class RepartitionTopicPurged implements TestCondition {
+        private TopicSizeVerifier verifier;
+
+        RepartitionTopicPurged(TopicSizeVerifier verifier) {
+            this.verifier = verifier;
+        }
+
+        @Override
+        public boolean conditionMet() {
+            time.sleep(purgeIntervalMs);
+
+            try {
+                Collection<DescribeLogDirsResponse.LogDirInfo> logDirInfo = adminClient.describeLogDirs(Collections.singleton(0)).values().get(0).get().values();
+
+                if (logDirInfo.isEmpty()) {
+                    return false;
+                } else {
+                    for (DescribeLogDirsResponse.LogDirInfo partitionInfo : logDirInfo) {
+                        DescribeLogDirsResponse.ReplicaInfo replicaInfo = partitionInfo.replicaInfos.get(new TopicPartition(REPARTITION_TOPIC, 0));
+                        if (replicaInfo != null && verifier.verify(replicaInfo.size)) {
+                            return true;
+                        }
+                    }
+                }
+                return false;
+            } catch (Exception e) {
+                return false;
+            }
+        }
+    }
+
+    @BeforeClass
+    public static void createTopics() throws InterruptedException {
+        CLUSTER.createTopic(INPUT_TOPIC, 1, 1);
+    }
+
+    @Before
+    public void setup() {
+        // create admin client for verification
+        Properties adminConfig = new Properties();
+        adminConfig.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        adminClient = AdminClient.create(adminConfig);
+
+        Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, APPLICATION_ID);
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(APPLICATION_ID).getPath());
+        streamsConfiguration.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_MS_CONFIG), purgeIntervalMs);
+        streamsConfiguration.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_BYTES_CONFIG), purgeSegmentBytes);
+        streamsConfiguration.put(StreamsConfig.producerPrefix(ProducerConfig.BATCH_SIZE_CONFIG), purgeSegmentBytes / 2);    // we cannot allow batch size larger than segment size
```

Comment:
Should we add a check to `StreamsConfig` for this and give a proper error message to users in case they miss configure?

Reply:
I think this is covered in `InternalTopicManager` unit tests?

Reply:
It's not about testing but about user experience. If one adds wrong configs and later sees the issue in the logs vs getting an exception at startup. Also, I am not sure what the logs error msg would look like and if it's clear to the user what the issues is. If we check this in `StreamsConfig` we can give a clear explanation. Of course, it's kind of a double check. Quite subjective and just an idea.

Reply:
I see what you mean now.. that's a good point, I'll add it.